code/game/blood.dm

```diff
@@ -1,84 +1,50 @@
 
 
-
-
-
-//to add a mob's dna info into an object's blood_DNA list.
-/atom/proc/transfer_mob_blood_dna(mob/living/L)
-	// Returns 0 if we have that blood already
-	var/new_blood_dna = L.get_blood_dna_list()
-	if(!new_blood_dna)
-		return 0
-	if(!blood_DNA)	//if our list of DNA doesn't exist yet, initialise it.
-		blood_DNA = list()
-	var/old_length = blood_DNA.len
-	blood_DNA |= new_blood_dna
-	if(blood_DNA.len == old_length)
-		return 0
-	return 1
-
-//to add blood dna info to the object's blood_DNA list
-/atom/proc/transfer_blood_dna(list/blood_dna)
-	if(!blood_DNA)
-		blood_DNA = list()
-
-	var/old_length = blood_DNA.len
-	blood_DNA |= blood_dna
-	if(blood_DNA.len > old_length)
-		return 1//some new blood DNA was added
-
-
 //to add blood from a mob onto something, and transfer their dna info and blood color
 /atom/proc/add_mob_blood(mob/living/M)
-	var/list/blood_dna = M.get_blood_dna_list()
-	if(!blood_dna)
-		return 0
 	var/b_color = M.get_blood_color()
 
-	return add_blood(blood_dna, b_color)
+	return add_blood(b_color)
 
 
 
 
 /////// add_blood ///////////////////
 
 //to add blood onto something, with blood dna info to include.
-/atom/proc/add_blood(list/blood_dna, b_color)
-	return 0
+/atom/proc/add_blood(b_color)
+	return FALSE
 
 
-/turf/add_blood(list/blood_dna, b_color)
+/turf/add_blood(b_color)
 	var/obj/effect/decal/cleanable/blood/splatter/B = locate() in src
 	if(!B)
 		B = new /obj/effect/decal/cleanable/blood/splatter(src)
 		if(b_color)
 			B.basecolor = b_color
 			B.color = b_color
-	B.transfer_blood_dna(blood_dna) //give blood info to the blood decal.
-	return 1 //we bloodied the floor
+	return TRUE //we bloodied the floor
 
 
-/obj/add_blood(list/blood_dna, b_color)
+/obj/add_blood(b_color)
 	if(flags_atom & NOBLOODY)
-		return 0
-	. = transfer_blood_dna(blood_dna)
+		return FALSE
 	if(b_color)
 		blood_color = b_color
+	return TRUE
 
 
-/obj/item/add_blood(list/blood_dna, b_color)
-	if(flags_atom & NOBLOODY)
-		return 0
-	var/blood_count = blood_DNA && blood_DNA.len
-	..()
-	if(!blood_count && !blood_overlay)//apply the blood-splatter overlay if it isn't already in there
+/obj/item/add_blood(b_color)
+	if(!..())
```

Comment:
```
. = ..()
if(!.)
```

Reply:
theres a hard return true at the end of the proc.